src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -2484,8 +2549,17 @@ public void watchFiles(File buildFile, File outputDirectory, File testOutputDire
                         }
                     }
                 }
+                if (!upstreamProjects.isEmpty()) { // process java compilation for upstream projects
```

Comment:
Should this check for null? e.g. `if (upstreamProjects != null && !upstreamProjects.isEmpty()) {`